### PR TITLE
Create changelogs for 3.14.1.0

### PR DESCRIPTION
**Pull request by Kleidukos**
This PR creates the appropriate changelogs for the 3.14.1.0 release.

### CODE REVIEW

**Comment by Kleidukos** [comment]
That's why I suggest to have release notes that are not the bare changelog: to be able to bring visibility to things that may otherwise be missed when people only read changelogs.

File: release-notes/Cabal-3.14.1.0.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+### Significant changes
+
+- Fix build ways for modules in executables [#10418](https://github.com/haskell/cabal/issues/10418) [#10419](https://github.com/haskell/cabal/pull/10419)
+
+  - Modules belonging to executables were being built in too many ways. For instance, if you
+  had configured to build profiled library files then your executable modules would also
+  be built profiled. Which was a regression in behaviour since `Cabal-3.12`.
+
+- Fix ./setup install command [#10416](https://github.com/haskell/cabal/issues/10416) [#10417](https://github.com/haskell/cabal/pull/10417)
```
Comment (ffaf1) [inline release-notes/Cabal-3.14.1.0.md:11]:
```suggestion
- Fix `./setup install` command [#10416](https://github.com/haskell/cabal/issues/10416) [#10417](https://github.com/haskell/cabal/pull/10417)
```

File: release-notes/cabal-install-3.14.1.0.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## cabal-install and cabal-install-solver 3.14.1.0 changelog
+
+- Fix a bug that causes `cabal init` to crash if `git` is not installed [#8478](https://github.com/haskell/cabal/issues/8478) [#10484](https://github.com/haskell/cabal/issues/10484) [#10486](https://github.com/haskell/cabal/pull/10486)
+
+  - `cabal init` tries to use `git config` to guess the user's name and email.
+    It no longer crashes if there is no executable named `git` on $PATH.
```
Comment (ffaf1) [inline release-notes/cabal-install-3.14.1.0.md:6]:
```suggestion
    It no longer crashes if there is no executable named `git` on `$PATH`.
```

**Review by ffaf1 — APPROVED** [review]

**Comment by Mikolaj** [comment]
I've had a look at the changelog and, unrelated, noticed that https://hackage.haskell.org/package/Cabal-3.14.0.0/changelog contains a duplicated entry for 3.12.0.0 and https://hackage.haskell.org/package/cabal-install-3.12.1.0/changelog is messed up as well. Could we fix both on 3.14 branch and on master? I can see the same problem in the repo at https://github.com/haskell/cabal/blob/master/Cabal/ChangeLog.md, while https://github.com/haskell/cabal/blob/master/cabal-install/changelog seems to be fixed. I haven't checked branch 3.14. I suppose this was at some point fixed on release branches and not forward-ported to master?

Related to your point: yes, I proposed to include both .0 and .1 changelogs in the .1 file, with proper headers and when that did not find support I begged to at least link to the older changelog, to make sure people understand there was a .0 release without the tool. We didn't do that. I think we should be courageous adding even logins preambles to changelogs. That's one of the advantage of having many changelog files instead of one big file --- with the big file, interspersing actual changelogs with rambling prose would be out of place, while here it's proper IMHO, since these are mixed changelog/release notes files.

**Comment by ulysses4ever** [comment]
Unrelated, would be great to convert the cabal-install changelog file (e.g. https://hackage.haskell.org/package/cabal-install-3.12.1.0/changelog) to proper markdown so that it’s rendered in a civil way (like, say, Cabal’s changelog, e.g. https://hackage.haskell.org/package/Cabal-3.14.0.0/changelog).

**Comment by ulysses4ever** [comment]
Having release notes is a fine idea, my issue is that it’s not something Hackage understands (unlike changelogs, which have a special place in the UI). So I’m unsure how you propose to distribute these “release notes”.

**Comment by Kleidukos** [comment]
@ulysses4ever https://blog.haskell.org as I said during one of our calls.

**Comment by ulysses4ever** [comment]
Mm, that could be fun, yes

**Pull request merged**